quantecon/markov/approximation.py

```diff
@@ -242,3 +245,198 @@ def _fill_tauchen(x, P, n, rho, sigma, half_step):
+    unit standard deviaiton; and :math:`{C}` is a volatility matrix.
+    Internally, from this process a sample time series of length
+    `sim_length` is produced, and with a cartesian grid as specified by
+    `grid_sizes` and `std_devs` a Markov chain is estimated that fits
+    the time series, where the states that are never visited under the
+    simulation are removed.
+
+    For a mathematical derivation check *Finite-State Approximation Of
+    VAR Processes: A Simulation Approach* by Stephanie Schmitt-Grohé and
+    Martín Uribe, July 11, 2010. In particular, we follow Schmitt-Grohé
+    and Uribe's method in contructing the grid for approximation.
+
+    Parameters
+    ----------
+    A : array_like(float, ndim=2)
+        An m x m matrix containing the process' autocorrelation
+        parameters. Its eigenvalues are assumed to have moduli 
+        bounded by unity.
+    C : array_like(float, ndim=2)
+        An m x r volatility matrix
+    grid_sizes : array_like(int, ndim=1), optional(default=None)
+        An m-vector containing the number of grid points in the
+        discretization of each dimension of x_t. If None, then set to
+        (10, ..., 10).
+    std_devs : float, optional(default=np.sqrt(10))
+        The number of standard deviations the grid should stretch in
+        each dimension, where standard deviations are measured under the
+        stationary distribution.
+    sim_length : int, optional(default=1_000_000)
+        The length of the simulated time series.
+    rv : optional(default=None)
+        Object that represents the disturbance term u_t. If None, then
+        standard normal distribution from numpy.random is used.
+        Alternatively, one can pass a "frozen" object of a multivariate
+        distribution from `scipy.stats`. It must have a zero mean and
+        unit standard deviation (of dimension m).
+    order : str, optional(default='C')
+        ('C' or 'F') order in which the states in the cartesian grid are
+        enumerated.
+    random_state : int or np.random.RandomState/Generator, optional
+        Random seed (integer) or np.random.RandomState or Generator
+        instance to set the initial state of the random number generator
+        for reproducibility. If None, a randomly initialized RandomState
+        is used.
+
+    Returns
+    -------
+    mc : MarkovChain
+        An instance of the MarkovChain class that stores the transition
+        matrix and state values returned by the discretization method,
+        in the following attributes:
+
+            `P` : ndarray(float, ndim=2)
+                A 2-dim array containing the transition probability
+                matrix over the discretized states.
+
+            `state_values` : ndarray(float, ndim=2)
+                A 2-dim array containing the state vectors (of dimension
+                m) as rows, which are ordered according to the `order`
+                option.
+
```

Comment:
Many thanks @oyamad. I think they are in the current branch after your merge. Please correct me if I am wrong.